GoogleTestAdapter/Core.Tests/Runners/CommandLineGeneratorTests.cs

```diff
@@ -309,6 +309,27 @@ public void GetCommandLines_ManyTestsWithSuites_BreaksUpLongCommandLinesCorrectl
             splittedTestsAsSet.Should().BeEquivalentTo(testsAsSet);
         }
 
+    [TestMethod]
+    [TestCategory(Unit)]
+    public void GetCommandLines_HugeNumberOfSuitesToBeExecutedFully_WontBreakCommandLineCreation()
+    {
+      var range = Enumerable.Range(0, 400);
+
+      var allTests = range.Select(idx => "MyTestSuiteWithSomeWhatLongNameIsStillHandledCorrectly" + idx + ".Test")
```

Comment:
You're right about that it is a little bit complicated. I made the test-suite-collection a little bit more complex to check the changed algorithm returns the right result set also for somewhat complicated collections (which I partially check by 
`commands.First().TestCases.Count(testCase => testCase.DisplayName.EndsWith("Test3")).Should().Be(1);`).
I create a test-suite-collection & test-set-to-execute, where some suites are fully tested, some are partially tested and where only a single suite with a 'Test3' is executed.
However for purely testing on the bug this is not neccessary (and maybe placed better in seperate tests) and your proposed test-set might be sufficient.

Reply:
I just realized that the most important corner cases of your code (e.g. "no suites", "one resulting command line only with suites", "one resulting command with suites and test cases") should be covered by the existing tests. Therefore, your test is probably fine - maybe add a tiny comment giving the rational for how you construct the test cases, and add some more assertions...

Reply:
I thought so. As I did not change the behavior or API there should be no need for any additional tests (except those that have been missing beforehand). Actually as I made my test more complicate than neccessary for reproducing the bug, I found a bug in my original bugfix (which was not found by any of the existing tests) ... still it might be a good idea to seperate concerns ...

Reply:
Finally improved documentation and added some addtional asserts :)